### PR TITLE
Build: Update grunt-jscs

### PR DESCRIPTION
**Pull request by gibson042**
cc @markelog

### CODE REVIEW

File: speed/speed.js [hunk 1]
```diff
@@ -557,29 +565,30 @@ function( Benchmark, document, selectors ) {
...
 				"The slowest is <i>" + slowest + " (" +
 				scores[ slowest ] + ")</i>.";
-			get("header").appendChild( elem );
+			get( "header" ).appendChild( elem );
 
 			// Add totals to table
 			i = firstTestedColumn();
-			while ( (elem = tds[ i++ ]) ) {
-				attr = elem.getAttribute("data-engine");
+			while ( ( elem = tds[ i++ ] ) ) {
```
Comment (gibson042) [inline speed/speed.js]: I'm _really_ not happy with introducing such whitespace into parenthesized expressions (as opposed to parameter/argument lists). Can anything be done about this?

Comment (timmywil) [inline speed/speed.js]: I agree with you, but I think the point is that we need consistency and the easiest way to get that now is by making the rules as simple as possible for jscs to enforce.

Comment (markelog) [inline speed/speed.js]: Well, if you remember, we three were outvoted on this matter, for me this very uncomfortable too :-(

Comment (dmethvin) [inline speed/speed.js]: I think the lack of spaces in the inner parens actually makes it clearer that they are doing grouping, so I am not a fan of extra spaces either. So is this mainly about making a simple rule ("There must be spaces around parens, except the left paren of a function call")?

Comment (gibson042) [inline speed/speed.js]: The rule that I want would apply to parenthesized expressions _specifically_ instead of only the broader class including function parameters and call expression arguments (all of which are wrapped in parentheses).

Comment (markelog) [inline speed/speed.js]: @gibson042 distinguishing grouping parentheses with other cases is pretty hard goal to accomplish :-(, otherwise i would did it a long time.

Nevertheless, you are very welcome to take a stab on it yourself

Comment (gibson042) [inline speed/speed.js]: It should be easier after the San Francisco ESTree concrete syntax discussion. But if you can point me at the relevant code in JSCS, I'll happily take a look sooner.

Comment (markelog) [inline speed/speed.js]: > It should be easier after the San Francisco ESTree concrete syntax discussion.

Not sure if there would be discussion about that :-(

> But if you can point me at the relevant code in JSCS, I'll happily take a look sooner.

Parentheses rules - (<a href="https://github.com/jscs-dev/node-jscs/blob/master/lib/rules/require-spaces-inside-parentheses.js">require</a> | <a href="https://github.com/jscs-dev/node-jscs/blob/master/lib/rules/disallow-spaces-inside-parentheses.js">disallow</a>)SpacesInsideParentheses

Comment (gibson042) [inline speed/speed.js]: https://github.com/jscs-dev/node-jscs/pull/1254

File: Gruntfile.js [hunk 2]
```diff
@@ -29,12 +29,11 @@ module.exports = function( grunt ) {
...
 
 		// See https://github.com/jquery/sizzle/wiki/Sizzle-Documentation#browsers
-
 		browsers.desktop = [
 			"bs_chrome-38", "bs_chrome-39",
 
-			"bs_firefox-24", "bs_firefox-31", // Firefox ESR
-			"bs_firefox-34", "bs_firefox-35",
+			// Old, ESR, (BrowserStack) latest minus one, (BrowserStack) latest
+			"bs_firefox-24", "bs_firefox-31", "bs_firefox-35", "bs_firefox-36",
```
Comment (markelog) [inline Gruntfile.js]: This probably deserves its own commit

**Comment by mgol** [comment]
This will need to land to make Sizzle buildable in Node 7 when it comes in 5 months as the current JSCS uses an old version of `graceful-fs` that will break there.

**Comment by timmywil** [comment]
It will also need a rebase to resolve conflicts.

**Comment by markelog** [comment]
> old version of graceful-fs that will break there.

Where? Could you provide a link?

**Comment by mgol** [comment]
@markelog 
```
$ node --version
v6.2.0
$ pm i graceful-fs@3
npm WARN deprecated graceful-fs@3.0.8: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
```

**Comment by markelog** [comment]
no-no, i meant "uses" part, where is it declared as dependency?

**Comment by mgol** [comment]
Ah, OK. It's easy to check. :)
```
$ npm ls graceful-fs
sizzle@2.3.1-pre /Users/mgol/Documents/projects/public/jquery/sizzle
├─┬ grunt@0.4.5
│ └─┬ glob@3.1.21
│   └── graceful-fs@1.2.3 
├─┬ grunt-jscs@0.6.2
│ └─┬ jscs@1.5.9
│   └─┬ glob@4.0.6
│     └── graceful-fs@3.0.8 
├─┬ karma@0.13.22
│ ├─┬ chokidar@1.5.1
│ │ └─┬ fsevents@1.0.12
│ │   └─┬ node-pre-gyp@0.6.25
│ │     └─┬ tar@2.2.1
│ │       └─┬ fstream@1.0.8
│ │         └── graceful-fs@4.1.3 
│ └── graceful-fs@4.1.4 
└─┬ karma-browserstack-launcher@1.0.0
  └─┬ browserstacktunnel-wrapper@1.4.2
    └─┬ unzip@0.1.11
      └─┬ fstream@0.1.31
        └── graceful-fs@3.0.8 
```
Wersions <4.0.0 are affected.

**Comment by markelog** [comment]
it seems we just using old `grunt-jscs` here, in any case never mind though, we will switch to eslint soon. I think we need to land this ASAP.

So i could start working on moving to eslint. Also, this

> This probably deserves its own commit

Is still stands, although i would just remove this changes, we need to update that list anyhow

**Comment by gibson042** [comment]
I'm closing this. An ESLint replacement would be welcomed, which I can do later if no one else is interested.

**Comment by markelog** [comment]
I would still merge it though, code style violations would be present either way, i was planing do the eslint migration too, right after core

**Comment by markelog** [comment]
Since you closed this pull despite my suggestion, i imagine you have your own plan how to do this without duplicating the effort here so i will leave you to it then
